### ISSUE TITLE
ERROR:memtest86+-5.01_4 INSTALL script failed to execute

### ISSUE THREAD

**Issue by b1scu1t** [void-linux/void-packages#10358]
### System

* xuname:  
 ` Void 4.14.108_1 x86_64 GenuineIntel notuptodate rF`
* package:  
  `memtest86+-5.01_4`

### Expected behavior
The `memtest86+-5.01_4` package would be installed/removed normally as the 
INSTALL and REMOVE scripts function normally.

### Actual behavior
The configure hook fails during installation with:
`[configure] INSTALL script failed to execute the post ACTION: Operation not permitted
`
It is also impossible to remove the package through `xbps-remove` as this happens:
```
Removing `memtest86+-5.01_4' ...
ERROR: memtest86+-5.01_4: [remove] REMOVE script failed to execute post ACTION: Operation not permitted
ERROR: memtest86+-5.01_4: failed to remove package: Operation not permitted
```
### Steps to reproduce the behavior
`# xbps-install -S memtest86+` 
OR
`# xbps-remove -R memtest86+`

I've got an `strace` of the `xbps-remove` process [here.](https://gist.github.com/b1scu1t/1147bc6ae3d0018fc5397fb5235a4a80#file-memtest86-_remove-log)

I apologize for the text dump...

**Comment by maxice8** [comment]
The INSTALL and REMOVE script will faill if grub-mkconfig are not found. memtest86+ should depend on it or the INSTALL/REMOVE scripts be adapted to it.

**Comment by maxice8** [comment]
That just reveals a bigger problem that we can't actually build memtest86+

**Comment by b1scu1t** [comment]
I've got `grub-2.02_4` installed, `grub-mkconfig` is there and the 'GRUB' hooks execute  with `xbps-reconfigure -f linux4.14`.
Forgive me, but where can I find the INSTALL and REMOVE scripts? 
Maybe I can check for permission issues or issues with $PATH?

**Comment by maxice8** [comment]
On Thu, Mar 28, 2019 at 10:46:21PM -0700, b1scu1t wrote:
> I've got `grub-2.02_4` installed, `grub-mkconfig` is there and the 'GRUB' hooks execute  with `xbps-reconfigure -f linux4.14`.
> Forgive me, but where can I find the INSTALL and REMOVE scripts? 
> Maybe I can check for permission issues or issues with $PATH?
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/issues/10358#issuecomment-477876020

srcpkgs/memtest86+/INSTALL

**Comment by b1scu1t** [comment]
Got an `strace`on the INSTALL script and I found this:
`stat("/boot/grub", 0x7ffc7e0d7c00)      = -1 ENOENT (No such file or directory)`

After running `lsblk`, it turns out that `/boot` isn't mounted (sda6).
```
 NAME                                          MAJ:MIN RM   SIZE RO TYPE  MOUNTPOINT
sda                                             8:0    0 465.8G  0 disk  
├─sda1                                          8:1    0   500M  0 part  
├─sda2                                          8:2    0  79.5G  0 part  /media/Windows
├─sda3                                          8:3    0     1K  0 part  
├─sda5                                          8:5    0 315.2G  0 part  /media/NTFS
├─sda6                                          8:6    0   400M  0 part  
└─sda7                                          8:7    0  70.2G  0 part  
  └─luks-a22d9680-2d8e-4dac-925e-71ed3e116daf 254:0    0  70.2G  0 crypt 
    ├─vg0-root                                254:1    0    18G  0 lvm   /
    ├─vg0-swap                                254:2    0     8G  0 lvm   [SWAP]
    └─vg0-home                                254:3    0  44.2G  0 lvm   /home
```
After mounting it with `sudo mount /dev/sda6 /boot`, it works fine.

Output of `sudo xbps-install -Sf memtest86+`

```
[*] Configuring unpacked packages
memtest86+-5.01_4: configuring ...
Generating grub configuration file ...
File descriptor 3 (socket:[1799297]) leaked on vgs invocation. Parent PID 573: /usr/bin/grub-probe
File descriptor 3 (socket:[1799297]) leaked on vgs invocation. Parent PID 573: /usr/bin/grub-probe
Found linux image: /boot/vmlinuz-4.14.108_1
Found initrd image: /boot/initramfs-4.14.108_1.img
File descriptor 3 (socket:[1799297]) leaked on vgs invocation. Parent PID 1038: /usr/bin/grub-probe
File descriptor 3 (socket:[1799297]) leaked on vgs invocation. Parent PID 1038: /usr/bin/grub-probe
Found memtest86+ image: /memtest86+
File descriptor 3 (socket:[1799297]) leaked on lvs invocation. Parent PID 1467: /bin/sh
Found Windows 7 on /dev/sda1
done
memtest86+-5.01_4: installed successfully.
```

**Comment by b1scu1t** [comment]
...And it turns out I didn't have the boot partition entry in `/ect/fstab`.
...
Sorry for wasting your time.
Should I close this?

**Comment by maxice8** [comment]
This is an actual bug in our INSTALL/REMOVE scripts, we are effectively stopping people from installing/removing memtest86+ if they don't have `grub` installled.

**Comment by maxice8** [comment]
please keep this open, this will be closed automatically once memtest86+ is fixed

**Comment by b1scu1t** [comment]
Understood, thanks!

**Comment by maxice8** [comment]
We believe this has been fixed in the latest version of memtest86+-5.01_5
